src/discovery/discovery_db.rs

```diff
@@ -723,9 +723,11 @@ mod tests {
       .create_publisher(&QosPolicies::qos_none())
       .unwrap();
     let dw2 = publisher2
-      .create_datawriter::<RandomData, CDRSerializerAdapter<RandomData, LittleEndian>>(&topic, None)
+      .create_datawriter::<RandomData, CDRSerializerAdapter<RandomData, LittleEndian>>(
+        &topic2, None,
```

Comment:
Passed in topic2 instead of topic, looks like a copy/paste type when this test was being created.